spring-kafka/src/main/java/org/springframework/kafka/config/KafkaListenerEndpointRegistrar.java

```diff
@@ -74,6 +79,25 @@ public KafkaListenerEndpointRegistry getEndpointRegistry() {
 		return this.endpointRegistry;
 	}
 
+	/**
+	 * Return the list of {@link HandlerMethodArgumentResolver}.
+	 * @return the list of {@link HandlerMethodArgumentResolver}.
+	 * @since 2.5
+	 */
+	public List<HandlerMethodArgumentResolver> getCustomMethodArgumentResolvers() {
+		return this.customMethodArgumentResolvers;
+	}
+
+	/**
+	 * Add custom methods arguments resolvers to {@link KafkaListenerAnnotationBeanPostProcessor}
+	 * Default empty list.
+	 * @param methodArgumentResolvers the list of {@link HandlerMethodArgumentResolver} to use.
+	 * @since 2.5
+	 */
+	public void setCustomMethodArgumentResolvers(List<HandlerMethodArgumentResolver> methodArgumentResolvers) {
```

Comment:
Let's consider varargs instead of list!
Much easier from end-user perspective.

Reply:
@artembilan to be honest I am not sure if this method should add elelments to existing list (maybe then I should change name to addCustomMethodArgumentResolvers) or assign to this filed completly new list. What do you think?

Reply:
Well, you are right: I'm not talking about *adding* - it is definitely has to be full replacement.
So, you get a varargs argument, clean up an existing `customMethodArgumentResolvers` state and add those new into it.